instance/models/mixins/openedx_database.py

```diff
@@ -234,6 +236,19 @@ def set_field_defaults(self):
         if not self.mongo_user:
             self.mongo_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
             self.mongo_pass = get_random_string(length=32)
+        if not self.rabbitmq_vhost:
+            def new_rabbitmq_user():
```

Comment:
This function is a replacement for other inline alternatives, where I would otherwise be looping over the two member fields I'm assigning (difficult to read), or cut/pasting the RabbitMQUser creation block (redundant and difficult to read).

I placed the function inline rather than as a method of the DatabaseMixin or elsewhere because I wanted to be clear that this function has no purpose anywhere else except for with the following two lines.

I reordered the placement of the fields to make this more clear. Do you think this makes sense? I'm also open to alternative patterns for creating these two users.